### PR TITLE
Refactor ccf model

### PR DESCRIPTION
**Pull request by manushak**
### Types of changes

- [x] Enhancement (project structure, spelling, grammar, formatting)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### A description of the changes proposed in the Pull Request

- Refactored CCF model
-

### CODE REVIEW

File: src/lib/ccf/index.ts [hunk 1]
```diff
@@ -12,426 +25,451 @@ import * as GCP_EMBODIED from './gcp-embodied.json';
...
+    this.computeInstances['azure'] = {};
   }
 
+  /**
+   * Calculates average of all instances.
+   */
   private calculateAverage(vendor: string, instanceList: KeyValuePair[]) {
-    let min = 0.0;
-    let max = 0.0;
+    let totalMin = 0.0;
```
Comment (narekhovhannisyan) [inline src/lib/ccf/index.ts:273]: if you got chance, please try with reduce and see if it works

File: src/lib/ccf/index.ts [hunk 2]
```diff
@@ -12,426 +25,451 @@ import * as GCP_EMBODIED from './gcp-embodied.json';
...
+  }
+
+  /**
+   * Calculates the average minimum and maximum watts consumption for AWS instances based on the provided architectures.
+   */
+  private calculateAWSAverageWatts(architectures: string[]) {
+    const awsInstance = this.instanceUsage['aws'];
+    let minWatts = 0.0;
+    let maxWatts = 0.0;
+    let count = 0;
```
Comment (narekhovhannisyan) [inline src/lib/ccf/index.ts:412]: if possible, apply reduce here too

File: src/lib/ccf/index.ts [hunk 3]
```diff
@@ -12,426 +25,451 @@ import * as GCP_EMBODIED from './gcp-embodied.json';
...
+
+  /**
+   * Calculates the consumption metrics (idle, 10%, 50%, 100%, minWatts, maxWatts) for a given compute instance.
+   */
+  private calculateAWSConsumption(instance: KeyValuePair, cpus: number) {
+    const architectures = this.getInstanceArchitectures(instance);
+    const {minWatts, maxWatts} = this.calculateAWSAverageWatts(architectures);
+
+    return {
+      idle: parseFloat(instance['Instance @ Idle'].replace(',', '.')),
```
Comment (narekhovhannisyan) [inline src/lib/ccf/index.ts:454]: you can convert parseFloat, replace to one reusable function and apply for all of props here

Comment (manushak) [reply src/lib/ccf/index.ts]: done

File: src/lib/ccf/index.ts (hunk 2 above)
Comment (manushak) [reply src/lib/ccf/index.ts]: done

File: src/lib/ccf/index.ts (hunk 1 above)
Comment (manushak) [reply src/lib/ccf/index.ts]: done

**Review by narekhovhannisyan — APPROVED** [review]

**Review by jmcook1186 — APPROVED** [review]
unit tests passing, running as expected from manifest file too.

**Pull request merged**